app/routes.py

```diff
@@ -1 +1,248 @@
+
+"""
+Get all tasks - sort by descending order if users specify to do so, 
+otherwise sort by ascending order by default
+"""
+
+@tasks_bp.route("",methods=["GET"])
+def check_all_tasks():
+    order_query = request.args.get("sort")
+    
+    if order_query == "desc":
+        tasks = Task.query.order_by(desc("title"))
+    else:
+        tasks = Task.query.order_by(asc("title"))
+        
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+        
+    return jsonify(tasks_response) 
+
+@tasks_bp.route("/<task_id>",methods=["GET"])
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+    
+    return {"task":task.to_dict()}
+    
+@tasks_bp.route("/<task_id>",methods=["PUT"])
+def update_a_task(task_id):
+    task = validate_model(Task, task_id)
+    
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+    
+    return {"task":task.to_dict()}
+    
+@tasks_bp.route("/<task_id>",methods=["DELETE"])
+def delete_a_task(task_id):
+    task = validate_model(Task, task_id)
+    
+    db.session.delete(task)
+    db.session.commit()
+    
+    return {
+        "details":f'Task {task.task_id} "{task.title}" successfully deleted'
+    }
+
+@tasks_bp.route("/<task_id>/mark_complete",methods=["PATCH"])
+def mark_complete_on_incompleted(task_id):
+    task = validate_model(Task, task_id)
+    
+    task.completed_at = date.today() 
+    
+    db.session.commit()
+    
+    data = {
+        "channel":"task-notifications",
+        "text":f"Someone just completed the task {task.title}",
+        "token":os.environ.get("BOT_API_KEY")
+    }
+    
+    requests.post("https://slack.com/api/chat.postMessage",data)
+    
+    return {
+        "task":{
+            "id":task.task_id,
+            "title":task.title,
+            "description":task.description,
+            "is_complete": True 
+        }
+    }
+
+@tasks_bp.route("/<task_id>/mark_incomplete",methods=["PATCH"])
+def mark_incomplete_on_completed(task_id):
+    task = validate_model(Task, task_id)
+    
+    task.completed_at = None 
+    
+    db.session.commit()
+    
+    return {
+        "task":{
+            "id":task.task_id,
+            "title":task.title,
+            "description":task.description,
+            "is_complete": False 
+        }
+    } 
+
+# CRUD for Goals
+
+goals_bp = Blueprint("goals",__name__,url_prefix="/goals")
+
+@goals_bp.route("",methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except:
+        abort(make_response({
+            "details":"Invalid data"
+        },400))
+    
+    db.session.add(new_goal)
+    db.session.commit() 
+    
+    response_body = {
+        "goal":{
+        "id":new_goal.goal_id,
+        "title":new_goal.title,
+        }
```

Comment:
This dictionary is created several times in the code. It would be good to DRY this up by having a helper function!